frontend/src/scenes/data-management/database/databaseTableListLogic.ts

```diff
@@ -54,186 +71,93 @@ export const databaseTableListLogic = kea<databaseTableListLogicType>([
 
                 return Object.values(database.tables)
             },
+            { resultEqualityCheck: objectsEqual },
         ],
         allTablesMap: [
-            (s) => [s.database],
-            (database): Record<string, DatabaseSchemaTable> => {
-                if (!database || !database.tables) {
-                    return {}
-                }
-
-                return Object.values(database.tables).reduce(
-                    (acc, cur) => {
-                        acc[cur.name] = database.tables[cur.name]
-                        return acc
-                    },
-                    {} as Record<string, DatabaseSchemaTable>
-                )
```

Comment:
roughly this was done repeatedly and made it hard (for me) to parse what was happening